stubs/wurlitzer/wurlitzer.pyi

```diff
@@ -0,0 +1,97 @@
+        encoding: str | None = ...,
+        bufsize: int | None = ...,
+    ) -> None: ...
+    def __enter__(self): ...
+    def __exit__(
+        self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: TracebackType | None
+    ) -> None: ...
+
+def dup2(a: int, b: int, timeout: float = 3) -> int: ...
+def sys_pipes(encoding: str = ..., bufsize: int | None = None) -> contextlib._GeneratorContextManager[tuple[TextIO, TextIO]]: ...
+
+# stubtest does not support overloaded context managers, hence the _GeneratorContextManager[Foo] return types.
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _STDOUT, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, None]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _PIPE, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, io.BytesIO]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _StreamErrT, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, _StreamErrT]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _STDOUT, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.StringIO, None]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _PIPE, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.StringIO, io.StringIO]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _StreamErrT, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.StringIO, _StreamErrT]]: ...
+@overload
+def pipes(
+    stdout: _StreamOutT, stderr: _STDOUT, encoding: str | None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[_StreamOutT, None]]: ...
+@overload
+def pipes(
+    stdout: _StreamOutT, stderr: _PIPE, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[_StreamOutT, io.BytesIO]]: ...
+@overload
+def pipes(
+    stdout: _StreamOutT, stderr: _PIPE, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[_StreamOutT, io.StringIO]]: ...
+@overload
+def pipes(
+    stdout: _StreamOutT, stderr: _StreamErrT, encoding: str | None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[_StreamOutT, _StreamErrT]]: ...
+@overload
+def pipes(
+    stdout: _PIPE | _StreamOutT = 3,
+    stderr: _STDOUT | _PIPE | _StreamErrT = 3,
+    encoding: str | None = ...,
+    bufsize: int | None = None,
+) -> _GeneratorContextManager[tuple[io.BytesIO | io.StringIO | _StreamOutT, io.BytesIO | io.StringIO | _StreamErrT | None]]: ...
+def sys_pipes_forever(encoding: str = ..., bufsize: int | None = None): ...
+def stop_sys_pipes() -> None: ...
+def load_ipython_extension(ip: _InteractiveShell) -> None: ...
+def unload_ipython_extension(ip: _InteractiveShell) -> None: ...
+
+__all__ = ["STDOUT", "PIPE", "Wurlitzer", "pipes", "sys_pipes", "sys_pipes_forever", "stop_sys_pipes"]
```

Comment:
Nit: It would be better to place `__all__` near the top of the file, similarly to how it is in wurlitzer's source code.

Reply:
Done in c34be49.

Reply:
Done in c34be49.